ovh/resource_cloud_project_volume.go

```diff
@@ -0,0 +1,208 @@
+		)
+		return
+	}
+
+	d.config = config
+}
+
+func (d *cloudProjectVolumeResource) Schema(ctx context.Context, req resource.SchemaRequest, resp *resource.SchemaResponse) {
+	resp.Schema = CloudProjectVolumeResourceSchema(ctx)
+}
+
+func (r *cloudProjectVolumeResource) Create(ctx context.Context, req resource.CreateRequest, resp *resource.CreateResponse) {
+	var data, responseData CloudProjectVolumeModelOp
+
+	// Read Terraform plan data into the model
+	resp.Diagnostics.Append(req.Plan.Get(ctx, &data)...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	endpoint := "/cloud/project/" + url.PathEscape(data.ServiceName.ValueString()) + "/region/" + url.PathEscape(data.RegionName.ValueString()) + "/volume"
+	if err := r.config.OVHClient.Post(endpoint, data.ToCreate(), &responseData); err != nil {
+		resp.Diagnostics.AddError(
+			fmt.Sprintf("Error calling Post %s", endpoint),
+			err.Error(),
+		)
+		return
+	}
+
+	err := r.WaitForVolumeCreation(ctx, r.config.OVHClient, data.ServiceName.ValueString(), responseData.Id.ValueString())
+	if err != nil {
+		resp.Diagnostics.AddError(
+			fmt.Sprintf("Error calling Operation"),
+			err.Error(),
+		)
+		return
+	}
+
+	res := &CloudProjectVolumeModelOp{}
+	endpoint = "/cloud/project/" + url.PathEscape(data.ServiceName.ValueString()) + "/operation/" + url.PathEscape(responseData.Id.ValueString())
+	if err := r.config.OVHClient.Get(endpoint, res); err != nil {
+		resp.Diagnostics.AddError(
+			fmt.Sprintf("Error calling Post %s", endpoint),
+			err.Error(),
+		)
+		return
+	}
+
+	resVol := &CloudProjectVolumeModelOp{}
+	endpoint = "/cloud/project/" + url.PathEscape(data.ServiceName.ValueString()) + "/region/" + url.PathEscape(data.RegionName.ValueString()) + "/volume/" + url.PathEscape(res.ResourceId.ValueString())
+	if err := r.config.OVHClient.Get(endpoint, resVol); err != nil {
+		resp.Diagnostics.AddError(
+			fmt.Sprintf("Error calling Post %s", endpoint),
+			err.Error(),
+		)
+		return
+	}
+
+	data.MergeWith(resVol)
+
+	// Save data into Terraform state
+	resp.Diagnostics.Append(resp.State.Set(ctx, &data)...)
+}
+
+func (d *cloudProjectVolumeResource) WaitForVolumeCreation(ctx context.Context, client *ovh.Client, serviceName, operationId string) error {
+	stateConf := &retry.StateChangeConf{
+		Pending: []string{"null", "in-progress", "created", ""},
+		Target:  []string{"completed"},
+		Refresh: func() (interface{}, string, error) {
+			res := &CloudProjectVolumeModelOp{}
+			endpoint := "/cloud/project/" + url.PathEscape(serviceName) + "/operation/" + url.PathEscape(operationId)
+			err := client.GetWithContext(ctx, endpoint, res)
+			if err != nil {
+				return res, "", err
+			}
+			return res, res.Status.ValueString(), nil
+		},
+		Timeout:    360 * time.Second,
+		Delay:      10 * time.Second,
+		MinTimeout: 3 * time.Second,
+	}
+
+	_, err := stateConf.WaitForStateContext(ctx)
```

Comment:
you should retrieve the last result instead like the following:

```
res, err := stateConf.WaitForStateContext(ctx)

return res, err
```

and use `res` in create func, so that you can remove lines 76-84.